workflow/controller/operator.go

```diff
@@ -1836,7 +1890,7 @@ func (woc *wfOperationCtx) executeContainer(nodeName string, templateScope strin
 	node := woc.getNodeByName(nodeName)
 	if node == nil {
 		node = woc.initializeExecutableNode(nodeName, wfv1.NodeTypePod, templateScope, tmpl, orgTmpl, opts.boundaryID, wfv1.NodePending)
-	} else if !isResubmitAllowed(tmpl) || !node.Pending() {
+	} else if tmpl.Synchronization == nil && !isResubmitAllowed(tmpl) || !node.Pending() {
```

Comment:
Could you explain this change? This is could have large consequences and I don't think it's intended